qiskit/pulse/instructions/acquire.py

```diff
@@ -76,24 +76,36 @@ def __init__(self,
             warnings.warn('The AcquireInstruction on multiple qubits, multiple '
                           'memory slots and multiple reg slots is deprecated. The '
                           'parameter "mem_slots" has been replaced by "mem_slot" and '
-                          '"reg_slots" has been replaced by "reg_slot"', DeprecationWarning, 3)
+                          '"reg_slots" has been replaced by "reg_slot"',
+                          DeprecationWarning, stacklevel=2)
 
         if not isinstance(channel, list):
             channels = [channel] if channel else None
         else:
             channels = channel
 
-        if mem_slot and not isinstance(mem_slot, list):
-            mem_slot = [mem_slot]
+        if mem_slot:
+            try:
+                mem_slot = list(mem_slot)
+            except TypeError:
+                mem_slot = [mem_slot]
         elif mem_slots:
             mem_slot = mem_slots
+        else:
+            mem_slot = []
 
         if reg_slot:
-            reg_slot = [reg_slot]
-        elif reg_slots and not isinstance(reg_slots, list):
-            reg_slot = [reg_slots]
+            try:
+                reg_slot = list(reg_slot)
+            except TypeError:
+                reg_slot = [reg_slot]
+        elif reg_slots:
+            try:
+                reg_slot = list(reg_slots)
+            except TypeError:
+                reg_slot = [reg_slots]
         else:
-            reg_slot = reg_slots
+            reg_slot = []
```

Comment:
Would you mind helping me understand what's changed, here?

Reply:
IIRC, if this block was triggered `reg_slots` is by definition None and `reg_slot` would be set to `None`. Causing an `operand` to be `None` when a list was expected for the deprecated case.